### PR TITLE
Cancel Orders

### PR DESCRIPTION
**Pull request by Francois-Gomis**

### CODE REVIEW

File: Controller/Payment/CancelOrder.php [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    public function execute()
+    {
+        $this->logger->info('In Cancel',[$this->getRequest()->getParams()]);
+        $paymentId = $this->getRequest()->getParam('pid');
+        try {
+            $this->almaPayment = $this->paymentValidation->getAlmaPayment($paymentId);
+        } catch (RequestError $e) {
+            $this->logger->info('Cancel order - get alma Payment error',[$e->getMessage()]);
```
Comment (syjust) [inline Controller/Payment/CancelOrder.php:63]: maybe a warning or error level for log here no ?

File: Controller/Payment/CancelOrder.php [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+        $this->logger = $logger;
+        $this->paymentValidation = $paymentValidation;
+        $this->orderFactory = $orderFactory;
+        $this->orderManagement = $orderManagement;
+    }
+
+    public function execute()
+    {
+        $this->logger->info('In Cancel',[$this->getRequest()->getParams()]);
```
Comment (syjust) [inline Controller/Payment/CancelOrder.php:58]: I'am not sure about this log necessity (according to following log info you had removed in `Helpers/Eligibliy` for example)

Comment (Francois-Gomis) [reply Controller/Payment/CancelOrder.php:58]: Remove no error log

File: Controller/Payment/CancelOrder.php (hunk 1 above)
Comment (Francois-Gomis) [reply Controller/Payment/CancelOrder.php:63]: change lvl to error

**Review by syjust-alma — APPROVED** [review]

**Review by syjust-alma — COMMENTED** [review]
bin/magento setup:upgrade should update default cancel_url

File: Setup/UpgradeData.php [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        $this->resourceConnection = $resourceConnection;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function upgrade(ModuleDataSetupInterface $setup, ModuleContextInterface $context)
+    {
+        $version = $context->getVersion();
+        if (version_compare($version, '2.3.2') < 0) {
```
Comment (syjust-alma) [inline Setup/UpgradeData.php:47]: `version_compare($version, '2.3.2', "<")` seems more readable for me

File: Setup/UpgradeData.php [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            $oldProcessedPath = $this->customerCancelUrl->getOldDefaultUrl();
+            $newProcessedPath = $this->customerCancelUrl->processValue('');
+
+            $tableName = $this->resourceConnection->getTableName('core_config_data');
+            $replace = ['value'=> $newProcessedPath];
+            $where = ['path = ?' => $path,'value = ?'=>$oldProcessedPath];
+            try {
+                $this->resourceConnection->getConnection()->update($tableName,$replace,$where);
+            } catch (Exception $e) {
+                $this->logger->info('UpgradeData Exception : ',[$e->getMessage()]);
```
Comment (syjust-alma) [inline Setup/UpgradeData.php:58]: maybe a `warning` or `error` log level here ?

File: Setup/UpgradeData.php (hunk 3 above)
Comment (Francois-Gomis) [reply Setup/UpgradeData.php]: Done

File: Setup/UpgradeData.php (hunk 4 above)
Comment (Francois-Gomis) [reply Setup/UpgradeData.php]: Done - error lvl

**Review by syjust-alma — APPROVED** [review]

**Pull request merged**